e2e/exec.go

```diff
@@ -24,8 +24,10 @@ type podExecResult struct {
 }
 
 func (r *podExecResult) dumpAll() {
-	r.dumpStdout()
-	r.dumpStderr()
+	if r != nil {
```

Comment:
Does this func need to be a pointer receiver? Does it mutate state?

Reply:
Good point, we can modify it to have a value receiver.

Reply:
yeah this makes sense, and also since the majority of this struct's fields are already pointers themselves so copies aren't too expensive

Reply:
changed to value semantics